smart-contracts/contracts/PublicLock.sol

```diff
@@ -2,6 +2,7 @@ pragma solidity 0.4.24;
 
 import "./interfaces/ILockPublic.sol";
 import "./interfaces/IUnlock.sol";
+import "openzeppelin-eth/contracts/introspection/ERC165.sol";
```

Comment:
Hey do you mind adding details about what ERC165 is for?

Reply:
Will do.

Reply:
@julien51 I added a comment in the contract as well, but ERC165 allows our contract to be queried to determine whether it implements a given interface. It's part of the ERC721 spec.
One great benefit it brings is the ability to test our interfaces themselves, and protect them against unconsidered changes. 
Eg: As it stands, our `PublicLock.sol` inherits the `ILockPublic.sol` interface. The compiler enforces that we implement each function speced in the interface. But, if we remove or comment out a function in the interface, our contract will still compile. Which is fine, but the next version/implementation of the contract no longer has the safety-net of the full interface.
With erc165 we can test that our contract implements a given interface directly, so that even if the interface is changed, we still have a test making sure we're compliant. If the test fails but the contract compiles, it can only mean that the interface has been modified.

Reply:
Very cool! Ok, please add a small comment in the code and/or link to the ERC165 page?

Reply:
@julien51 Done. Also, this is now closed and replaced by #778